<launch>    
    
  <!-- <node name="SLAM_sensor" pkg="SLAM_sensor" type="SLAM_sensor" output="screen"/>
 -->
  

  <node ns="/" name="stereo_image_proc" pkg="stereo_image_proc" type="stereo_image_proc" output="screen">
    <remap from="left/image_raw" to="/cam1/image_raw" />
    <remap from="right/image_raw" to="/cam0/image_raw" />
    <remap from="left/camera_info" to="/cam1/camera_info" />
    <remap from="right/camera_info" to="/cam0/camera_info" />

    <param name="min_disparity" value="3" />
    <param name="disparity_range" value="128"/>
    <param name="speckle_size" value="900"/>
    <param name="speckle_range" value="6"/>

    <param name="queue_size" value="30" />
    <!-- <param name="approximate_sync" value="false"/> -->
  </node>

  <node pkg="fovis_ros" type="fovis_stereo_odometer" name="fovis" output="screen">
    <remap from="stereo" to="/" />
    <remap from="image" to="image_rect" />
    <remap from="//left/camera_info" to="/cam1/camera_info" />
    <remap from="//right/camera_info" to="/cam0/camera_info" />
    <param name="approximate_sync" value="False" />
    <param name="odom_frame_id" value="map" />
    <param name="publish_tf" value="True" />
    <param name="base_link_frame_id" value="imu4" />

    <!-- <param name="stereo_max_disparity" value="64" /> -->
    <param name="inlier_max_reprojection_error" type="string" value="2" />

    
  </node>

  <node pkg="tf" type="static_transform_publisher" name="static_publisher" args="0 0 0 1.5708 0 -1.5708 imu4 cam 10">
  </node>

  <node pkg="tf" type="static_transform_publisher" name="static_publisher2" args="0 0 0 0 3.1415 0 cam cam0 10">
  </node>

  <node pkg="tf" type="static_transform_publisher" name="static_publisher3" args="0 0 0 0 0 0 imu4 imu 10">
  </node>

  <!-- <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf" output="screen">
    <param name="output_frame" value="odom_fixed"/>
    
    <param name="base_footprint_frame" value="imu4"/>
    <param name="freq" value="30.0"/>
    <param name="sensor_timeout" value="20.0"/>
    <param name="odom_used" value="false"/>
    <param name="imu_used" value="true"/>
    <param name="vo_used" value="true"/>
    <param name="debug" value="True"/>
    <param name="self_diagnose" value="True"/>


    <remap from="imu_data" to="/imu/data_fixed" />
    <remap from="vo" to="/viso/odometry" />
    
  </node>

  <node pkg="imu_filter_madgwick" type="imu_filter_node" name="imu_filter" output="screen">
   <remap from="/imu/data_raw" to="/imu/data" />
   <remap from="/imu/mag" to="/magnetic" />
   <remap from="imu/data" to="imu/data_fixed" />

   <param name="use_mag" value="true" />
   <param name="fixed_frame" value="imu" />
   <param name="publish_tf" value="false" />
   <param name="gain" value="0.1" />
  </node> -->

  <!-- <node name="octomap_server_node" pkg="octomap_server" type="octomap_server_node" output="screen">
    <param name="sensor_model/max_range" value="8" />
    <param name="resolution" value="0.05" />
    <param name="latch" value="false" />
    <remap from="cloud_in" to="/points2" />
  </node> -->



</launch>
